app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -44,21 +53,24 @@ def on_success(_status, options)
   def build_batch
     batch = Sidekiq::Batch.new
     batch.description = batch_description
-    batch_success_callback(batch) { batch.jobs { delete_associations(object) } }
+    batch_callbacks(batch) { batch.jobs { delete_associations(object) } }
     batch
   end
 
   def batch_description
     "Deleting #{object.class.name} [##{object.id}]"
   end
 
-  def batch_success_callback(batch)
-    options = { 'object_global_id' => object.to_global_id, 'caller_worker_hierarchy' => caller_worker_hierarchy }
-
-    batch.on(:success, self.class, options)
+  def batch_callbacks(batch)
+    %i[success complete].each { |name| batch.on(name, self.class, callback_options) }
     yield
     bid = batch.bid
-    on_success(bid, options) if Sidekiq::Batch::Status.new(bid).total.zero?
+    if Sidekiq::Batch::Status.new(bid).total.zero?
+      on_complete(bid, callback_options)
+    else
+      info "DeleteObjectHierarchyWorker#batch_success_callback retry job with the hierarchy of workers: #{caller_worker_hierarchy}"
+      retry_job wait: 5.minutes
```

Comment:
i was thinking about max retry count, but there's no API for that in the sidekiq and at the end i didn't want to go that way.. 

raising an error wouldn't change anything since we would need to rescue that anyway.. but we could send an event to a bugsnag to know that something's going on.. wdyt?

Reply:
+ `on_complete` is being called even if some workers have failed.. so this worker shouldn't be in the queue forever.

Reply:
OK for logging the error

Reply:
I think it is a good idea to have the bugnag in case it ever happens so we can see it and control it 👍 

Another option (which doesn't fully convince me 🤔 ) would be changing this line for:
`DeletePlainObjectWorker.set(wait: 10.minutes).perform_later(...)`
So it would call directly the other Worker in 10 minutes no matter if it has finished or not 🤔

Reply:
that could destroy the main object without waiting for associations to be destroyed..

ok, let's add bugsnag reporting in there.